src/api/hooks/useGetTransaction.ts

```diff
@@ -0,0 +1,16 @@
+import {Types} from "aptos";
+import {useQuery} from "react-query";
+import {getTransaction} from "../../api";
+import {ResponseError} from "../../api/client";
+import {useGlobalState} from "../../GlobalState";
+
+export function useGetTransaction(txnHashOrVersion: string) {
+  const [state, _setState] = useGlobalState();
+
+  const {data, status} = useQuery<Types.Transaction, ResponseError>(
```

Comment:
Why is the rest of the query result discarded?

Reply:
For the snackbar response we only need `data` and `status`. We can probably add more in the future when they're needed.

Reply:
it might worth returning the full result so then other components (https://github.com/aptos-labs/explorer/blob/main/src/pages/layout/Search.tsx#L256 , https://github.com/aptos-labs/explorer/blob/main/src/pages/Transactions/Transactions.tsx#L133) can use this hook. then each component can use whatever data it needs

Reply:
made the change